pootle/apps/pootle_store/migrations/0037_set_unit_changed_with.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-02-17 11:53
+from __future__ import unicode_literals
+
+import logging
+import time
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+
+def set_unit_changed_with(apps, schema_editor):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    UnitChange = apps.get_model("pootle_store.UnitChange")
+    unit_changes = {}
+    last_unit = 0
+    sub_data = subs.order_by("unit_id", "-creation_time", "-id").values_list(
+        "unit_id", "type")
+    for unit_id, sub_type in sub_data.iterator():
+        if unit_id != last_unit:
+            unit_changes[sub_type] = unit_changes.get(sub_type, [])
```

Comment:
This is getting a list that is larger with each iteration and assigning it again, why not:

```python
            if sub_type not in unit_changes:
                unit_changes[sub_type] = []
```